Add support for optional Console plugin dependencies (#15183)
openshift/console: frontend/packages/console-dynamic-plugin-sdk/README.md

`version` must be [semver](https://semver.org/) compliant version string.

### Exposed modules

Dynamic plugins can expose modules representing plugin code that can be referenced, loaded and executed
at runtime. A separate [webpack chunk](https://webpack.js.org/guides/code-splitting/) is generated for
each entry in the `exposedModules` object. Exposed modules are resolved relative to the plugin's webpack
`context` option.

### Dependencies

> Dynamic plugins can declare dependency on specific Console versions and/or other plugins. This metadata

Author: @opayne1 

> Can we remove the "can" here? I think with the word "can" it implies plugins sometimes declare dependency and sometimes they don't.

Correct, plugins *might* declare dependencies (whether on Console itself or on other plugins) but it's not mandatory. This is why it was worded as "can".

What wording do you suggest here?

> Dynamic plugins can declare dependency on specific Console versions and/or other plugins. This metadata

Reviewer: Why don't we use "might" here then? So 
```suggestion
Dynamic plugins might declare dependency on specific Console versions and other plugins. This metadata
```
